### PR TITLE
Add multi-tag functionality

### LINKED ISSUE #844: Publish with multiple tags
### Description

I'd like to publish an image with pack, with multiple tags. For instance, `dfreilich/pack:latest` and `dfreilich/pack:v0.0.15`. 

### Proposed solution

Add a flag `--tag` to allow for inputting multiple tags. This should work in cases where we call the creator, and can pass in the `-tag` flag. https://github.com/buildpacks/spec/blob/main/platform.md#creator

### Additional context
- [x] This feature should be documented somewhere

### PR DESCRIPTION
**Pull request by dwillist**
Signed-off-by: dwillist <dthornton@vmware.com>

## Summary
add support for a `--tag` build flags to an arbitrary number of additional tags. A pack build will push the resulting image to these location in addition to `name`

Ex `pack build <name> ... --tag <extra-location-1> --tag <extra-location-2> ...`

## Documentation

- Should this change be documented?
    - [x] Yes, see #___ (todo)
    - [ ] No

## Related

Resolves #844

### CODE REVIEW

File: acceptance/acceptance_test.go [hunk 1]
```diff
@@ -739,6 +738,7 @@ func testAcceptance(
 							runImageMirror,
 						)
 						assert.Nil(err)
+
```
Comment (dfreilich) [inline acceptance/acceptance_test.go:741]: nit: Remove
(Totally no need to do that if there's no need for a further commit, just a very very very tiny thing that caught my eye at the top)

File: acceptance/acceptance_test.go [hunk 2]
```diff
@@ -761,6 +761,29 @@ func testAcceptance(
...
+					when("additional tags", func() {
+						var additionalRepoName string
+
+						it.Before(func() {
+							additionalRepoName = fmt.Sprintf("%s_additional", repoName)
+						})
+						it("pushes image to additional tags", func() {
+							h.SkipUnless(t,
+								pack.Supports("build --tag"),
+								"--network flag not supported for build",
```
Comment (dfreilich) [inline acceptance/acceptance_test.go:774]:
```suggestion
								"--tag flag not supported for build",
```

File: acceptance/acceptance_test.go [hunk 3]
```diff
@@ -1595,6 +1618,68 @@ func testAcceptance(
...
+							var additionalRepoName string
+
+							it.Before(func() {
+								additionalRepo = fmt.Sprintf("%s_additional", repo)
+								additionalRepoName = fmt.Sprintf("%s_additional", repoName)
+							})
+							it("creates additional tags on the registry", func() {
+								h.SkipUnless(t,
+									pack.Supports("build --tag"),
+									"--network flag not supported for build",
```
Comment (dfreilich) [inline acceptance/acceptance_test.go:1633]:
```suggestion
									"--tag flag not supported for build",
```

File: internal/commands/build.go [hunk 4]
```diff
@@ -165,6 +167,7 @@ func buildCommandFlags(cmd *cobra.Command, buildFlags *BuildFlags, cfg config.Co
 	cmd.Flags().StringArrayVar(&buildFlags.Volumes, "volume", nil, "Mount host volume into the build container, in the form '<host path>:<target path>[:<mode>]'."+multiValueHelp("volume"))
 	cmd.Flags().StringVarP(&buildFlags.DefaultProcessType, "default-process", "D", "", `Set the default process type. (default "web")`)
 	cmd.Flags().StringVar(&buildFlags.Policy, "pull-policy", "", `Pull policy to use. Accepted values are always, never, and if-not-present. (default "always")`)
+	cmd.Flags().StringSliceVarP(&buildFlags.AdditionalTags, "tag", "t", nil, `Additional tags to push the output image to`)
```
Comment (dfreilich) [inline internal/commands/build.go:170]:
```suggestion
	cmd.Flags().StringSliceVarP(&buildFlags.AdditionalTags, "tag", "t", nil, "Additional tags to push the output image to."+multiValueHelp("tag"))
```

**Review by dfreilich — APPROVED** [review]
boom

File: acceptance/acceptance_test.go (hunk 1 above)
Comment (dwillist) [reply acceptance/acceptance_test.go:741]: Ya this paired with the last extra function I missed in the `acceptance_test.go` file make me think they should be linted during `make verify`.

**Review by jromero — APPROVED** [review]

**Pull request merged**